### PR TITLE
CRM-19752 - Unit Test for issue CRM-19752

### PR DESCRIPTION
**Pull request by vedantrathore**
Extracted a function getSummaryQuery(), added unit Tests for checking the function with ACL's enabled and ACL's disabled.

---

 * [CRM-19752: Slow query created by financial type acls](https://issues.civicrm.org/jira/browse/CRM-19752)

### CODE REVIEW

**Comment by vedantrathore** [comment]
@eileenmcnaughton could you please review this?

**Comment by colemanw** [comment]
@civicrm-builder add to whitelist

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
If you click through to https://test.civicrm.org/job/CiviCRM-Core-PR/13768/console (click on the console link from the jenkins report) you can see there is a merge issue, perhaps because there are 4 similar looking commits

To fix it do a rebase - ie

git pull --rebase origin master

to push rebased changes back up to your repo you will need to use the -f switch (force)

**Comment by vedantrathore** [comment]
@eileenmcnaughton I have rebased and squashed them into a single commit, I think this should work now.

**Comment by eileenmcnaughton** [comment]
OK we are homing in on it - you need to fix some more style issues. You could try using https://github.com/civicrm/coder although TBH I used jenkins a lot to check style.

When you fix those style issues if you use git commit --amend (& then use the force switch while pushing) you can adjust your existing commit rather than creating a new one

**Comment by vedantrathore** [comment]
Ok cool, I'll see the jenkins report, will civilint do the same job?

**Comment by colemanw** [comment]
Yes, it's generally a good idea to run civilint before committing. It 
doesn't check every file in the tree, only the ones with uncommitted 
changes.

I wrote a handy little script for frequent contributors, it will run 
civilint, checkout a branch, make a commit and then open a pull-request 
all in one command:

https://gist.github.com/colemanw/01dae2a2937f6f39c2300b701636b4a8

**Comment by vedantrathore** [comment]
Okay, thanks @colemanw  :smile:

**Comment by vedantrathore** [comment]
@eileenmcnaughton All tests passed, I think It's mergeable now. :)

**Comment by mlutfy** [comment]
jenkins, test this please

File: CRM/Contact/BAO/Query.php [hunk 1]
```diff
@@ -4895,17 +4895,10 @@ public function &summaryContribution($context = NULL) {
...
-                      ON civicrm_contribution.id = li.contribution_id AND
-                         li.entity_table = 'civicrm_contribution' AND li.financial_type_id NOT IN (" . implode(',', array_keys($financialTypes)) . ") ";
-    }
-    else {
-      $where .= " AND civicrm_contribution.financial_type_id IN (0)";
-    }
 
+    $query = $this->getSummaryQuery($where, $from);
+    $where = $query[0];
+    $from  = $query[1];
```
Comment (eileenmcnaughton) [inline CRM/Contact/BAO/Query.php]: this needs to be .=

Comment (eileenmcnaughton) [reply CRM/Contact/BAO/Query.php]: nope it doesnt'

**Comment by eileenmcnaughton** [comment]
Sorry about the huge delay on reviewing this. I just took a look & my reservation was that I want to take this change a bit further. However, this PR moves us in the right direction & adds patches so I'm going to merge & then add more as a follow up

**Pull request merged**